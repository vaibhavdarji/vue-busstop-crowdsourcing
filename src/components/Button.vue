<template>
  <v-btn 
    :type="btnType"
    :loading="isLoading"
    v-bind="{ [`${Constants.BUTTON.SIZE}`]: true }"
    :color="Constants.BUTTON.BG_COLOR"
    :class="[`${Constants.BUTTON.COLOR}--text`]"
    :disabled="disabled"
    @click="onClickBtn"
  >
    <slot></slot>
  </v-btn>
</template>

<script>
import Constants from '@/constants';
export default {
  name:'CustomButton',
  props: {
    isLoading: {
      type: Boolean,
      default: false
    },
    btnType: {
      type: String,
      default: 'button' 
    },
    disabled: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {};
  },
  computed: {
    Constants () {
      return Constants;
    }
  },
  methods: {
    onClickBtn () {
      this.$emit('onClick');
    } 
  }
}
</script>

