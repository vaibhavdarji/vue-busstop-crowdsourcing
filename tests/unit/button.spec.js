import { shallowMount } from "@vue/test-utils";
import Button from "@/components/Button.vue";
import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);

describe("Button.vue", () => {
  it("should match snapshot", () => {
    const wrapper = shallowMount(Button, {
      Vue
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});