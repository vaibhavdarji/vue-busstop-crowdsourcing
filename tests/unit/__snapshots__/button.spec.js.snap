// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should match snapshot 1`] = `
<v-btn-stub
  activeclass="v-btn--active"
  class="white--text"
  color="#2979FF"
  large="true"
  tag="button"
  type="button"
/>
`;
